Add Insights Coverage Report doc (#956)
saucelabs/sauce-docs: docs/insights/coverage.md

---
id: coverage
title: Measuring Your Test Coverage
sidebar_label: Coverage
description: See a breakdown of the browser and device variations covered by your tests.
---
import useBaseUrl from '@docusaurus/useBaseUrl';

> The Coverage report allows Org Admins to view and download the breakdown of browsers and mobile devices the tests across their organization have run against.

Reviewer: Can we add here one or two sentences describing the value of this tool? Not sure if the docs are the place for it. But if we do something along the lines of:
- make better decisions based on data 
- use data to inform your testing strategy and coverage based on your market needs

Maybe Alistair can shed some more light on the value proposition?

You must have [Org Admin](/basics/acct-team-mgmt/managing-user-info/#user-roles) privileges for your organization in order to view and download coverage data.
:::

## Access the Coverage Reports

1. Click the **Insights** tab to expand its submenu.
1. Click **Coverage** (only visible to Org Admin users).
1. Specify the type of coverage you wish to view by selecting the **Devices** or **Browsers** tab.
1. Specify whether you wish to see coverage data for **Virtual Cloud** tests (desktop web-apps, mobile emulators and simulators) or **Real Devices** (physical devices located in our data centers and connected to our platform).
1.Use the [filters](scope.md#using-filters-to-adjust-the-scope-of-your-data) to show results for a particular user's tests and/or a specific time period.

Once you have completed your selections, the report shows a breakdown of all tests that have been run for the specified user during the specified time period, providing the exact number and total duration of tests run in each environment with a bar graph for easy visualization.

<img src={useBaseUrl('img/insights/coverage.png')} alt="Coverage Report" width="900"/>

:::note Zero Coverage Data

> The coverage reports are not intended to identify test environments for which you have no coverage. That is to say, the report does not show the set of browsers or devices that are supported by Sauce Labs but for which the number of sessions run is zero.

Reviewer: I don't understand this paragraph.
Can you elaborate what is meant by "identify test environments"?
What does "but for which the number of sessions run is zero"

:::

## Sorting the Data

By default, coverage is shown in descending order based on the number of tests run, but you can modify the sort order by clicking on any of the column headings and then toggle using the ascending/descending icon.

<img src={useBaseUrl('img/insights/sorting.png')} alt="Edit Sort Order" width="600"/>

As the image shows, sorting by the **Name** column sorts the data alphabetically by the name of the browser or device, allowing you to quickly discern how much coverage a particular environment has in your testing.

## Downloading Your Report

You can download your report locally as a CSV file by clicking the **Export CSV** link.

<img src={useBaseUrl('img/insights/export-coverage.png')} alt="Export to CSV File" width="400"/>

> The resulting file includes the data set as it is filtered (by environment, owner, and time period) in the default order, regardless of any sorting you have specified in the UI.

Reviewer: is it data set or dataset?

---
id: coverage
title: Measuring Your Test Coverage
sidebar_label: Coverage
description: See a breakdown of the browser and device variations covered by your tests.
---
import useBaseUrl from '@docusaurus/useBaseUrl';

> The Coverage report allows Org Admins to view and download the breakdown of browsers and mobile devices the tests across their organization have run against. Having insight into your coverage data helps companies:

Reviewer 2: The Coverage report, accessible to Org Admins only, lists the browsers and mobile devices on which your organization has tested.

Author: I like the way you have phrased this, but I am going to leave it as is because I really want to emphasize the ability to view AND download the report.

* See at a glance whether their testing is in sync with the environments in which it is most often accessed
* Design their test strategy to keep up with newly released browser and device versions

> * Utilize their concurrency to spread testing across test environments for greater efficiency and comprehensiveness

Reviewer 2: Add bullet (pulled from line 32):

* Evaluate how evenly tests are being distributing across browsers and devices

:::note Org Admins Only
You must have [Org Admin](/basics/acct-team-mgmt/managing-user-info/#user-roles) privileges for your organization in order to view and download coverage data.
:::

> ## Access the Coverage Reports

Reviewer 2: Change Access to Accessing -- consistency with other H2 headers.

1. Click the **Insights** tab to expand its submenu.
1. Click **Coverage** (only visible to Org Admin users).
1. Specify the type of coverage you wish to view by selecting the **Devices** or **Browsers** tab.
1. Specify whether you wish to see coverage data for **Virtual Cloud** tests (desktop web-apps, mobile emulators and simulators) or **Real Devices** (physical devices located in our data centers and connected to our platform).

> 1.Use the [filters](scope.md#using-filters-to-adjust-the-scope-of-your-data) to show results for a particular user's tests and/or a specific time period.

Reviewer 2: Add space after **1.**

Apply the **Owner** and time period [filters](scope.md#using-filters-to-adjust-the-scope-of-your-data) to show data for a particular user's tests and/or a specific time period.

> Once you have completed your selections, the report shows a breakdown of all tests that have been run for the specified user during the specified time period, providing the exact number and total duration of tests run in each environment with a bar graph for easy visualization.

Author: ```suggestion
Once Kim has completed her selections, the report shows a breakdown of all tests that have been run for the specified user during the specified time period, providing the exact number and total duration of tests run in each environment with a bar graph for easy visualization.
```

<img src={useBaseUrl('img/insights/coverage.png')} alt="Coverage Report" width="900"/>

> :::note Browsers and Devices with No Coverage

Reviewer 2: I would keep this super simple. Something like:

:::note
Browsers and devices not used by your selected **Owner** during your selected time period will not appear in the Coverage report.
:::

If you have not run any tests against certain browsers and devices during the period shown, those browsers and devices are not shown in the report. The report is intended to show how evenly you are distributing your tests across browsers and devices on which you are actually testing, not to alert you to browser and device versions you have not covered.
:::

## Sorting the Data

> By default, coverage is shown in descending order based on the number of tests run, but you can modify the sort order by clicking on any of the column headings and then toggle using the ascending/descending icon.

Reviewer 2: _recommendation_:  By default, coverage is shown in descending order based on the number of tests run. You can modify the sort order by clicking on any of the column headings and then toggling the ascending/descending icon.
